packages/medusa/src/api/routes/admin/products/list-products.ts

```diff
@@ -0,0 +1,176 @@
+import { Type } from "class-transformer"
+import {
+  IsArray,
+  IsBoolean,
+  IsEnum,
+  IsNumber,
+  IsOptional,
+  IsString,
+  ValidateNested,
+} from "class-validator"
+import * as _ from "lodash"
+import { identity } from "lodash"
+import { defaultAdminProductFields, defaultAdminProductRelations } from "."
+import { DateComparisonOperator } from "../../../../types/common"
+import { validator } from "../../../../utils/validator"
+
+/**
+ * @oas [get] /products
+ * operationId: "GetProducts"
+ * summary: "List Product"
+ * description: "Retrieves a list of Product"
+ * x-authenticated: true
```

Comment:
Nit: Could add query params to the oas description, alike done [here](https://github.com/medusajs/medusa/pull/805).